### PR TITLE
Reject modify command when an element for activation is unknown

### LINKED ISSUE #9976: Reject the modification command if an element for activation is unknown
## Description

The modification command can contain activation instructions for different elements. The elements are identified by their element ids. 

If at least one element id is unknown (i.e. the process of the given instance doesn't contain an element with the given id) then the command should be rejected. 

Blocked by 
- #9639

### PR DESCRIPTION
**Pull request by korthout**
## Description

The modification command can contain activation instructions for different elements. The elements are identified by their element ids. If at least one element id is unknown (i.e. the process of the given instance doesn't contain an element with the given id) then the command is rejected as `NOT_FOUND`.

The reason for the rejection contains a reference to the Bpmn Process ID and all of the unknown element IDs. For example:
`Expected to activate element but no element found in process 'my_process' for element id(s): 'B', 'C'`

## Related issues

closes #9976 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java [hunk 1]
```diff
@@ -117,6 +129,27 @@ public void processRecord(
...
+            .map(ProcessInstanceModificationActivateInstructionValue::getElementId)
+            .filter(targetElementId -> process.getProcess().getElementById(targetElementId) == null)
+            .collect(Collectors.toSet());
+    if (!unknownElementIds.isEmpty()) {
+      final String reason =
+          String.format(
+              ERROR_MESSAGE_TARGET_ELEMENT_NOT_FOUND,
+              BufferUtil.bufferAsString(process.getBpmnProcessId()),
+              String.join("', '", unknownElementIds));
+      return Optional.of(new Rejection(RejectionType.NOT_FOUND, reason));
```
Comment (saig0) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java:147]: :wrench: We could align the rejection type and use `INVALID_ARGUMENT` instead.

In [CreateProcessInstanceProcessor](https://github.com/camunda/zeebe/blob/ac67c70ed765ad4a010c569516fce656916dd061/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java#L181-L197), we reject the command with the type `INVALID_ARGUMENT` if one element id of an activation instruction is unknown. We use `NOT_FOUND` only if the target entity (i.e. the process instance in this case) doesn't exist.

**Review by saig0 — APPROVED** [review]
@korthout looks good :+1: 

I have one suggestion regarding the rejection type. Please have a look.

File: engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java (hunk 1 above)
Comment (korthout) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java:147]: That's a good idea. Thanks 👍

**Comment by korthout** [comment]
bors merge

**Comment by korthout** [comment]
Failed due to flaky test: https://github.com/camunda/zeebe/issues/10169

**Comment by remcowesterhoud** [comment]
bors retry

**Pull request merged**